sql/driver/driver_test.go

```diff
@@ -288,3 +290,125 @@ func TestInsecure(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func strToInts(t *testing.T, results resultSlice) []int {
+	values := make([]int, 0, 3)
+	for _, result := range results[1:] {
+		if len(result) != 1 {
+			t.Fatalf("wrong number of columns: %d", len(result))
+		}
+		strVal := result[0]
+		if strVal == nil {
+			t.Fatal("unexpected nil value")
+		}
+		value, err := strconv.Atoi(*strVal)
+		if err != nil {
+			t.Fatal(err)
+		}
+		values = append(values, value)
+	}
+	return values
+}
+
+// concurrentIncrements starts two Goroutines in parallel, both of which
+// read the integer stored at the other's key, increment and update their own.
+// It checks that the outcome is serializable, i.e. exactly one of the
+// two Goroutines (the later write) sees the previous write by the other.
+func concurrentIncrements(db *sql.DB, t *testing.T) {
+	var wgEnd sync.WaitGroup
+	wgEnd.Add(2)
+	for i := 0; i < 2; i++ {
+		go func(i int) {
+			// Read the other key, write key i.
+			readKey := (i + 1) % 2
+			writeKey := i
+			defer wgEnd.Done()
```

Comment:
The test in the `client` package waits until the other goroutine is running. This test doesn't. Seems easy to add back in the `wgStart` stuff.